### PR TITLE
Business logic

### LINKED ISSUE #46: GET /api/v1/merchants/most_revenue?quantity=x
returns the top x merchants ranked by total revenue

### LINKED ISSUE #48: GET /api/v1/merchants/revenue?date=x
returns the total revenue for date x across all merchants

assume the dates provided match the format of a standard ActiveRecord timestamp

### LINKED ISSUE #51: GET /api/v1/merchants/:id/favorite_customer
returns the customer who has conducted the most total number of successful transactions.

### PR DESCRIPTION
**Pull request by dionew1**
### Description of changes:

- Favorite Customer added
- Complete favorite customer
- Complete top merchant/most revenue
- All Merchants Revenue for date

Complete Total merchant revenue for date

### CODE REVIEW

File: spec/requests/api/v1/merchants_request_spec.rb [hunk 1]
```diff
@@ -173,5 +173,120 @@
...
+      expect(response).to be_success
+      expect(favorite_customer_response["first_name"]).to eq(customer.first_name)
+      expect(favorite_customer_response["last_name"]).to eq(customer.last_name)
+    end
+
+    it 'returns top merchants when quantity 1 for merchant' do
+      merchant      = create(:merchant)
+      merchant2     = create(:merchant)
+      invoice       = create(:invoice, merchant: merchant)
+      invoice2      = create(:invoice, merchant: merchant2)
```
Comment (anlewis) [inline spec/requests/api/v1/merchants_request_spec.rb:202]: We should try to condense these test setups down the line...if we have time

File: app/models/merchant.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class Merchant < ApplicationRecord
   has_many :invoices
   has_many :items
+  has_many :customers, through: :invoices
 
-  default_scope { order(:id) }
+  # default_scope { order(:id) }
```
Comment (anlewis) [inline app/models/merchant.rb:6]: Did you ever uncomment this? Or should we just remove/change it?

**Review by anlewis — APPROVED** [review]

File: spec/requests/api/v1/merchants_request_spec.rb (hunk 1 above)
Comment (dionew1) [reply spec/requests/api/v1/merchants_request_spec.rb:202]: I agree

File: app/models/merchant.rb (hunk 2 above)
Comment (dionew1) [reply app/models/merchant.rb:6]: I think we will need to remove it so it does not interfere with ordering, unless you know that we need it.

**Pull request merged**